### ISSUE TITLE
Package is broken in Flutter 3.20+

### ISSUE THREAD

**Issue by mit-mit** [ABausG/home_widget#225]
Flutter landed a breaking change to the `ViewConstraints` class:
https://github.com/flutter/flutter/pull/140918/files#diff-5b359cf09d9f2b7ac85f97ab408aea9962ecac296e11120930464904b8833dd8

As a result, `home_widget` no longer passes analysis in Flutter 3.20 and later:

```
$ flutter analyze
Analyzing home_widget...

  error • The named parameter 'size' isn't defined • lib/home_widget.dart:176:11 • undefined_named_parameter

1 issue found. (ran in 4.7s)
```

```
$ flutter --version
Flutter 3.20.0-1.2.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 1751123cde (4 days ago) • 2024-02-21 22:06:22 -0800
Engine • revision 299e852cb9
Tools • Dart 3.4.0 (build 3.4.0-99.1.beta) • DevTools 2.31.0
```

**Comment by mit-mit** [comment]
I'm happy to send a PR to fix this -- the change is pretty trivial. But what branch should this PR be for (it will be some months before this change is in Flutter stable; it's currently only in Flutter beta)? I think the best option might be to make this change in a `beta` branch or similar, and then publish a [pre-release](https://dart.dev/tools/pub/publishing#publishing-prereleases) version of this package?

**Comment by ABausG** [comment]
Thanks for flagging this! As far as I can see this PR will fix it https://github.com/ABausG/home_widget/pull/221

Would you be able to answer the following question for me regarding pub versions:
- I assume that with this change I should adjust the flutter version to also match `3.20.0-1.2.pre` should this then be a breaking change in the version of home_widget or is pub able to determine to resolve to the last supported version?

**Comment by mit-mit** [comment]
Hi @ABausG any chance we could have this published? There are some demo apps I'd like to fix that need the update.

**Comment by ABausG** [comment]
I plan to do new releases this weekend! Sorry for the delay. I'll let you know in here once the new versions are there!

**Comment by ABausG** [comment]
I just released 0.6.0 with the flutter constraint as you suggested. 

https://pub.dev/packages/home_widget/versions/0.6.0
